recipes/IWSLT22_lowresource/hparams/train_w2v2_mbart_st.yaml

```diff
@@ -0,0 +1,186 @@
+# ############################################################################
+# Model: E2E ST with wav2vec 2.0 encoder
+# Encoder: wav2vec 2.0 + linear projection
+# Decoder: Transformer
+# Tokens: unigram (1000)
+# losses: NLL
+# Training: Tamasheq-French corpus
+# Author:  Marcely Zanon Boito, 2022
+# ############################################################################
+
+# Seed needs to be set at top of yaml, before objects with parameters are made
+seed: 1337 #7777
+__set_seed: !!python/object/apply:torch.manual_seed [!ref <seed>]
+debug: False
+output_folder: !ref output_w2v2_mbart/<seed>
+save_folder: !ref <output_folder>/save
+train_log: !ref <output_folder>/train_log.txt
+wer_file: !ref <output_folder>/wer.txt
+bleu_file: !ref <output_folder>/bleu.txt
+
+# root data folder points to 17h version inside the github folder (IWSLT2022_Tamasheq_data/taq_fra_clean/)
+root_data_folder: !PLACEHOLDER # e.g., /users/hnguyen/IWSLT2022_Tamasheq_data/taq_fra_clean
+# data folder is the place where the json files will be stored prior to training
+data_folder: !ref <root_data_folder>/json_version/
+lang: "fr" #for the BLEU score detokenization
+target_lang: "fr_XX" # for mbart initialization
+
+
+# URL for the HuggingFace model we want to load (BASE here)
+wav2vec2_hub: LIA-AvignonUniversity/IWSLT2022-tamasheq-only
+wav2vec2_folder: !ref <save_folder>/wav2vec2_checkpoint
+
+# wav2vec 2.0 specific parameters
+wav2vec2_frozen: False
+
+# Training parameters
+number_of_epochs: 500
+lr: 0.001
+lr_wav2vec: 0.0001
+lr_mbart: 0.0001
+batch_size: 2
+test_batch_size: 1
+gradient_accumulation: 6
+valid_search_interval: 4
+loss_reduction: batchmean
+ckpt_interval_minutes: 15 # save checkpoint every N min
+
+# Data sorting parameters: sorting_debug_duration replaces sorting_min_duration in debug mode
+sorting: ascending
+
+epoch_counter: !new:speechbrain.utils.epoch_loop.EpochCounter
+    limit: !ref <number_of_epochs>
+
+dataloader_options:
+    batch_size: !ref <batch_size>
+    num_workers: 4
+
+test_dataloader_options:
+    batch_size: !ref <test_batch_size>
+    num_workers: 4
+
+# Feature parameters (W2V2 etc)
+features_dim: 768 # base wav2vec output dimension, for large replace by 1024
+
+#projection for w2v
+enc_dnn_layers: 1
+enc_dnn_neurons: 1024 #256
+
+# Transformer
+activation: !name:torch.nn.GELU
+
+# Outputs
+label_smoothing: 0.1
+pad_index: 1      # pad_index defined by mbart model
+bos_index: 250008 # fr_XX bos_index defined by mbart model
+eos_index: 2
+
+# Decoding parameters
+# Be sure that the bos and eos index match with the BPEs ones
+min_decode_ratio: 0.0
+max_decode_ratio: 0.25
```

Comment:
oof why the max_decode_ratio is so low ?

Reply:
I don't recall exactly the reason. I guess because the target text sequence length (#words) is usually 1/4 of the source speech sequence (#frames). The BLEU report seems to agree with this. The total number of hypotheses' words is close to that of the references.

Reply:
Yes but the model should be able to recognize when to stop right ? E.g. by producing the eos token ?

Reply:
Could you please try 1.0 vs 0.25 and report back the results here ?

Reply:
I can but I have a strong feeling that it doesn't change make. In fact, I did try several values of this (0.5, 1.0) and the results were not different. Indeed as you said, the model learns to produce eos most the time.

Reply:
@Adel-Moumen : I confirm here that setting this value to 1.0 doesn't change much the BLEU scores (close to say that it doesn't change at all)